src/vos/vos_aggregate.c

```diff
@@ -1811,19 +1825,32 @@ vos_aggregate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	case VOS_ITER_AKEY:
 		rc = vos_agg_akey(ih, entry, agg_param, acts);
 		break;
-	case VOS_ITER_SINGLE:
-		rc = vos_agg_sv(ih, entry, agg_param, acts);
-		break;
 	case VOS_ITER_RECX:
-		rc = vos_agg_ev(ih, entry, agg_param, acts);
+	case VOS_ITER_SINGLE:
+		if (type == VOS_ITER_SINGLE)
+			rc = vos_agg_sv(ih, entry, agg_param, acts);
+		else
+			rc = vos_agg_ev(ih, entry, agg_param, acts);
 		if (rc == -DER_CSUM || rc == -DER_TX_BUSY) {
-			/* Abort current evtree aggregation only */
-			D_DEBUG(DB_EPC, "Abort evtree aggregation "DF_RC"\n",
+			D_DEBUG(DB_EPC, "Abort value aggregation "DF_RC"\n",
 				DP_RC(rc));
 
-			*acts |= VOS_ITER_CB_ABORT;
-			if (rc == -DER_CSUM)
-				agg_param->ap_csum_err = true;
+			if (rc == -DER_TX_RESTART)
```

Comment:
This check will be always false since you have checked "if (rc == -DER_CSUM || rc == -DER_TX_BUSY)" at L1834.

Reply:
You are right.  I guess I never actually hit this case because the case that was causing issues was that aborted entries were getting ignored entirely.  I'll fix.

Reply:
I reworked this a bit